File: python/src/estruturas/repetitivas/crescente.py
""" Problema "crescente"
 Leia uma quantidade indeterminada de duplas de valores inteiros X e Y.
 Escreva para cada X e Y uma mensagem que indique se estes valores foram digitados
 em ordem crescente ou decrescente.
 O programa deve finalizar quando forem digitados dois valores iguais.
"""


def input_numbers(message: str) -> tuple[int, int]:
    print(message)
    return int(input()), int(input())


def crescente() -> None:
    a, b = input_numbers("Digite dois números inteiros (a) e (b):")

    while a != b:
        if a > b:
            print("DECRESCENTE!")
        elif a < b:
            print("CRESCENTE!")
        else:
            break

        a, b = input_numbers("Digite outro dois números inteiros (a) e (b):")

    else:
        print("IGUAIS!")


if __name__ == "__main__":
    crescente()

File: java/src/estruturas/repetitivas/Crescente.java
package estruturas.repetidas;

import java.util.Scanner;

/**
 * Problema "crescente"
 * Leia uma quantidade indeterminada de duplas de valores
 * inteiros X e Y. Escreva para cada X e Y uma
 * mensagem que indique se estes valores foram digitados
 * em ordem crescente ou decrescente. O programa deve
 * finalizar quando forem digitados dois valores iguais.
 */
public class Crescente {
  /**
   * The entry point of application.
   *
   * @param args the input arguments
   */
  public static void main(String[] args) {
    Scanner sc = new Scanner(System.in);

    int a;
    int b;

    do {
      System.out.println("Digite dois números inteiros (a) e (b):");

      a = sc.nextInt();
      b = sc.nextInt();

      if (a > b) {
        System.out.println("DECRESCENTE!");
      } else if (a < b) {
        System.out.println("CRESCENTE!");
      }

    } while (a != b);

    System.out.println("IGUAIS!");

    sc.close();
  }
}